jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/DefaultRule.java

```diff
@@ -243,11 +246,35 @@ private JExpression getDefaultSet(JType fieldType, JsonNode node) {
 
     }
 
-    private JExpression getDefaultEnum(JType fieldType, JsonNode node) {
+    private JType getParameterType(JDefinedClass fieldType, String methodName){
 
-        JInvocation invokeFromValue = ((JClass) fieldType).staticInvoke("fromValue");
-        invokeFromValue.arg(node.asText());
+        for (JMethod method : fieldType.methods()) {
+            if (EnumRule.FROM_VALUE_METHOD_NAME.equals(method.name())
+                    && (method.mods().getValue() & JMod.PUBLIC) == JMod.PUBLIC
+                    && (method.mods().getValue() & JMod.STATIC) == JMod.STATIC) {
+                final JType[] type = method.listParamTypes();
+                if (type.length != 1)
+                    throw new GenerationException("Factory method '" + EnumRule.FROM_VALUE_METHOD_NAME + "' should has only one parameter in " + fieldType);
+                return type[0];
+            }
+        }
+        throw new GenerationException("Factory method '" + EnumRule.FROM_VALUE_METHOD_NAME + "' is not found in " + fieldType);
+
+    }
 
+    private JExpression getDefaultEnum(JDefinedClass fieldType, JsonNode node) {
+
+        JInvocation invokeFromValue = fieldType.staticInvoke(EnumRule.FROM_VALUE_METHOD_NAME);
+        JType paramType = getParameterType(fieldType, EnumRule.FROM_VALUE_METHOD_NAME).unboxify();
+        if (paramType == paramType.owner().INT) {
```

Comment:
@s13o Sorry, this conversation is getting very confused.

I originall added a comment here:
https://github.com/joelittlejohn/jsonschema2pojo/pull/707#issuecomment-287763365

What I'm saying is: there is no need to duplicate `if`/`else` blocks from `getDefaultValue` into `getDefaultEnum`. The `getDefaultEnum` method can **call** `getDefaultValue` if I am not mistaken. Inside `getDefaultEnum` (line 269) you have `JType paramType`, you have `JsonNode node` and you need a `JExpression` result. You have everything you need on line 269 to call `getDefaultValue` instead of implementing a duplicate `if`/`else` block.